test/linter/test-obsolete.js

```diff
@@ -0,0 +1,110 @@
+  'http.headers.Cache-Control.stale-if-error',
+  'http.headers.Feature-Policy.layout-animations',
+  'http.headers.Feature-Policy.legacy-image-formats',
+  'http.headers.Feature-Policy.oversized-images',
+  'http.headers.Feature-Policy.unoptimized-images',
+  'http.headers.Feature-Policy.unsized-media',
+  'svg.attributes.presentation.color-rendering',
+  'svg.elements.view.zoomAndPan',
+];
+
+function neverImplemented(support) {
+  for (const s in support) {
+    let data = support[s];
+    if (!Array.isArray(data)) data = [data];
+    for (const d of data) if (d.version_added) return false;
+  }
+  return true;
+}
+
+/**
+ * @param {*} browsers
+ * @param {*} support
+ * @returns 'warning' | 'error' | false
+ */
+function implementedAndRemoved(browsers, support) {
+  let result = 'error';
+  const warningTime = new Date().getTime() - 2 * 365 * 24 * 60 * 60 * 1000;
+  const errorTime = new Date().getTime() - 2.5 * 365 * 24 * 60 * 60 * 1000;
+  for (const browser in support) {
+    let data = support[browser];
+    if (!Array.isArray(data)) data = [data];
+    for (const d of data) {
+      // Feature is still supported
+      if (!d.version_removed) return false;
+      const releaseDate = new Date(
+        browsers[browser].releases[d.version_removed].release_date,
+      ).getTime();
+      // Feature was recently supported, no need to show warning
+      if (warningTime < releaseDate) return false;
+      // Feature was supported sufficiently recently to not show an error
+      if (errorTime < releaseDate) result = 'warning';
+    }
+  }
+  return result;
+}
+
+/**
+ * @param {string} name
+ * @param {CompatStatement} data The data to test
+ * @param {Logger} logger
+ * @param {boolean} shouldFail
+ * @returns {void}
+ */
+function processData(logger, data, browsers, path) {
+  if (data && data.support) {
+    const { support, status } = data;
+    const shouldFail = exceptions.includes(path);
+
+    const abandoned = status && status.standard_track === false;
+    const rule1Fail = abandoned && neverImplemented(support);
+    if (rule1Fail && !shouldFail) {
+      logger.error(
+        chalk`feature was never implemented in any browser and the specification has been abandoned.`,
+      );
+      return;
+    }
+
+    const rule2Fail = implementedAndRemoved(browsers, support);
+    if (rule2Fail && !shouldFail) {
+      logger[rule2Fail](
+        chalk`feature was implemented and has since been removed from all browsers dating back two or more years ago.`,
+      );
+      return;
+    }
+
+    if (!rule1Fail && !rule2Fail && shouldFail) {
+      logger.error(
+        chalk`Please remove this file from list of exceptions: ${path}`,
+      );
+    }
+  }
+}
+
+export default {
+  name: 'Obsolete',
+  description: 'Test for osolete data in each support statement',
+  scope: 'feature',
+  check(logger, { data, path }) {
+    processData(logger, data, browsers, path.full);
+  },
+  internals: {
+    neverImplemented,
+  },
```

Comment:
Rather than export an `internals` property on our default export, I think that we should instead provide a named export of the function and import it that way.

```suggestion
```